.github/workflows/osrm-backend.yml

```diff
@@ -34,12 +36,38 @@ jobs:
         node-version: 16
     - run: node --version
     - run: npm --version
+    - name: Prepare environment
+      shell: bash
+      run: |
+        PACKAGE_JSON_VERSION=$(node -e "console.log(require('./package.json').version)")
+        echo PUBLISH=$([[ "${GITHUB_REF:-}" == "refs/tags/v${PACKAGE_JSON_VERSION}" ]] && echo "On" || echo "Off") >> $GITHUB_ENV
     - run: npm install --ignore-scripts
     - run: npm link --ignore-scripts
     - uses: microsoft/setup-msbuild@v1.1
     - name: Build
       run: |
         .\scripts\ci\windows-build.bat
+    - name: Run node tests
+      shell: bash
+      run: |
+        ./lib/binding/osrm-datastore.exe test/data/ch/monaco.osrm
+        node test/nodejs/index.js | faucet
+    - name: Build Node package
+      if: ${{ env.PUBLISH == 'On' }}
+      run: bash ./scripts/ci/node_package.sh
```

Comment:
Checked that it works via manually setting `PUBLISH=On`

Reply:
Probably worth removing the conditional like you've done in #6360?

Reply:
Agree, will do 👍🏻